### ISSUE TITLE
Save as indexed image

### ISSUE THREAD

**Issue by basxto** [PureAsbestos/Image-palettizer#3]
Saving images as 8bit PNG/BMP with given palette would make sense

**Comment by PureAsbestos** [comment]
Hi there! You can save your image as either PNG or BMP by adding the appropriate extension to the image before saving it.

**Comment by PureAsbestos** [comment]
Oh, I get what you are trying to say. Save it in "palette mode." I'll see what I can come up with.

**Comment by PureAsbestos** [comment]
Hmm... I seem to be able to get it to work right with PNG and mostly right with GIF, but BMP eludes me

**Comment by PureAsbestos** [comment]
GIF always gives a palette with 256 entries, I'm going to leave it at that

**Comment by PureAsbestos** [comment]
It works with PNG, so that will be included in the next update

**Comment by PureAsbestos** [comment]
If you find a way to do this for BMP in imageio, I'd be more than happy to implement it

**Comment by PureAsbestos** [comment]
PNGs will now be saved as paletted, see v3.2.3

**Comment by basxto** [comment]
It looks like it’s not possible, at least not how I had it in mind.
The resulting PNG does not have palette defined in the GPL file, but an optimized variation of it.

**Comment by PureAsbestos** [comment]
Yeah, the palette gets reordered. I think the root problem is that there's no way to take a 2D array of integers and save it as an indexed image directly. Right now I have to convert from indexed to RGB, then when I save it I can have the image "quantized", but since I've already done that it ends up with the correct palette.

**Comment by basxto** [comment]
Pillow on the other hand has some methods that look interesting:
```
Image.convert(mode=None, matrix=None, dither=None, palette=0, colors=256)
Image.remap_palette(dest_map, source_palette=None)
```
https://pillow.readthedocs.io/en/stable/reference/Image.html

**Comment by PureAsbestos** [comment]
If Pillow is indeed capable of this functionality, perhaps an issue should be submitted to imageio to take advantage of it (seeing as it already relies on Pillow).
